cm-gateway/gateway-api-install/main.tf

```diff
@@ -0,0 +1,23 @@
+terraform {
+  required_providers {
+    null = {
+      source  = "hashicorp/null"
+      version = "~> 3.1.0"
+    }
+  }
+}
+
+resource "null_resource" "gateway-api" {
+  provisioner "local-exec" {
+    environment = {
+      KUBECONFIG = pathexpand(var.kubeconfig_path)
+    }
+
+    command = "kubectl apply -f ./crd/gateway-api.yaml"
+  }
+
+  provisioner "local-exec" {
+    when    = destroy
+    command = "kubectl delete -f ./crd/gateway-api.yaml"
+  }
+}
```

Comment:
I think it would be better to use a `kubernetes_manifest` or a similar more Terraform-native approach to install these CRDs. We could, for example use [`yamlencode`](https://www.terraform.io/language/functions/yamlencode) to encode the contents of a files with CRD definitions. I haven't tried using `yamlencode` with a file with multiple resources - perhaps we'd need to split the CRDs into separate files or use something like https://peterdaugaardrasmussen.com/2021/12/01/terraform-create-resources-from-kubernetes-yaml-files-with-multiple-configurations/
Let me know what you think.

I think using a Terraform-native mechanism instead of wrapping `kubectl apply` is that  this way we actually use Terraform features better, i.e Terraform can actually compare if the actual state equals teh desired state etc.

Reply:
Thanks for your suggestions, I will try it.